pydra/tasks/mrtrix3/utils.py

```diff
@@ -0,0 +1,155 @@
+        ),
+        (
+            "coord",
+            attr.ib(
+                type=ty.List[float],
+                metadata={
+                    "argstr": "-coord",
+                    "help_string": "extract data at the specific coordinatest",
+                },
+            ),
+        ),
+        (
+            "vox",
+            attr.ib(
+                type=ty.List[float],
+                metadata={
+                    "argstr": "-vox",
+                    "help_string": "change the voxel dimensions",
+                },
+            ),
+        ),
+        (
+            "axes",
+            attr.ib(
+                type=ty.List[int],
+                metadata={
+                    "argstr": "-axes",
+                    "help_string": "specify the axes that will be used",
+                },
+            ),
+        ),
+        (
+            "scaling",
+            attr.ib(
+                type=ty.List[float],
+                metadata={
+                    "argstr": "-scaling",
+                    "help_string": "specify the data scaling parameter",
+                },
+            ),
+        ),
+        (
+            "export_grad",
+            attr.ib(
+                type=str,
+                metadata={
+                    "argstr": "-export_grad_mrtrix",
+                    "position": 3,
+                    "help_string":  "export new gradient files in mrtrix3 format"
+                },
+            ),
+        ),
+        (
+            "export_json",
+            attr.ib(
+                type=str,
+                metadata={
+                    "argstr": "-json_export",
+                    "help_string": "export image headet to JSON file"
+                },
+            ),
+        ),
+    ],
+    bases=(MRTrix3BaseSpec,),
+)
+
+MRConvertOutputSpec = SpecInfo(
+    name = "MRConvertOutputs",
+    fields=[
+        (
+            "out_file",
+            attr.ib(
+                type=File,
+                metadata={
+                    "help_string": "output image",
+                    "output_file_template": "dwi.mif"
+                },
+            ),
+        ),
+        (
+            "out_bfile",
+            attr.ib(
+                type=File,
+                metadata={
+                    "help_string": "output .b gradient file in mrtrix3 format",
+                    "output_file_template": "dwi.b"
+                },
+            ),
+        ),
+        (
+            "out_json",
+            attr.ib(
+                type=File,
+                metadata={
+                    "help_string": "output JSON file of image header",
+                    "output_file_template": "dwi.json"
+                },
+            ),
+        ),
+    ],
+    bases=(ShellOutSpec,),
+)
+
+
+#  Create task for pydra
+MRConvert =  ShellCommandTask(
+    name='MRConvert',
+    executable="mrconvert",
+    in_file="/Users/xxie/sample_data/tinnitus2/sub-MINN294/ses-01/dwi/sub-MINN294_ses-01_run-1_dwi.nii",
```

Comment:
i think the error is coming from the fact that this file doesn't exist on the server.

at present as long as there is a dummy folder containing this bids structure, you should be able to use it, or since this looks more like an actual test, you could use datalad to fetch a simple dataset with dwi and then actually run the test using pytest.  

and install mrtrix3 into the environment as well.  for this you can either use neurodocker or install mrtrix3 through other mechanisms.

pinging @djarecka who is on vacation but should be able to help after she is back on wed.

Reply:
ahh, didn't realize I left a local directory in there... 😨 oops, and yeah I can try reconfiguring the github actions like you suggested, thanks!

Reply:
@satra I've created a test directory for the mrtrix3 tasks and created a test for the task, it seems like I will need to add installation instructions for mrtrix3 to github actions for tests to be executed.

Should I be looking to install mrtrix3 and all its dependencies or create a docker for the github actions to use?  Not sure what's the best approach here since mrtrix3 is a big & complicated thing =/

Reply:
docker may be the way to go. especially since pydra allows executing a task using docker.